examples/blas/run_time_dispatching/level3/CMakeLists.txt

```diff
@@ -0,0 +1,75 @@
+
+#===============================================================================
+# Copyright 2020-2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+#
+# SPDX-License-Identifier: Apache-2.0
+#===============================================================================
+
+# NOTE: user need to set env var SYCL_DEVICE_FILTER to use runtime example without specifying backend in CMake
+# $ENV{SYCL_DEVICE_FILTER}
+
+
+# Build object from all example sources
+set(BLAS_RT_SOURCES "gemm_usm")
+
+# Set up for the right backend for run-time dispatching examples
+# If users build more than one backend (i.e. mklcpu and mklgpu, or mklcpu and CUDA), they may need to
+# overwrite SYCL_DEVICE_FILTER in their environment to run on the desired backend
```

Comment:
Can we state, if SYCL_DEVICE_FILTER is not set, example will be run on every device?

Reply:
if SYCL_DEVICE_FILTER is not set, example will be run on one default device (like mklcpu)

Reply:
Oh, I think I misunderstood. So with ctest RT examples will run on all available devices because we should test on all devices. 
```
  # Register example as ctest
  foreach(device_filter ${DEVICE_FILTERS})
    add_test(NAME ${domain}/EXAMPLE/RT/${blas_rt_source}/${device_filter} COMMAND example_${domain}_${blas_rt_source})
    set_property(TEST ${domain}/EXAMPLE/RT/${blas_rt_source}/${device_filter} PROPERTY
      ENVIRONMENT LD_LIBRARY_PATH=${CMAKE_BINARY_DIR}/lib:$ENV{LD_LIBRARY_PATH}
      ENVIRONMENT SYCL_DEVICE_FILTER=${device_filter})
  endforeach(device_filter)
```
But if users just run the binary themselves, then it will be run on one default device if  SYCL_DEVICE_FILTER is not set

Reply:
ok, makes sense now. Thanks